snippets/c.snippets

```diff
@@ -140,21 +197,39 @@ snippet td
 	typedef ${1:int} ${2:MyCustomType};
 # struct
 snippet st
+	/*! \struct $1 
+	 *  \brief ${3:Brief struct description}
+	 * 
+	 *  ${4:Detailed description} 	
+	 */
 	struct ${1:`vim_snippets#Filename('$1_t', 'name')`} {
-		${2:/* data */}
-	}${3: /* optional variable list */};
+		${2:Data} /*!< ${4:Description} */
+	}${5: /* optional variable list */};
 # typedef struct
 snippet tds
+	/*! \struct $2 
+	 *  \brief ${5:Brief struct description}
+	 * 
+	 *  ${6:Detailed description} 	
+	 */
 	typedef struct ${2:_$1 }{
-		${3:/* data */}
+		m_${3:Data} /*!< ${4:Description} */
 	} ${1:`vim_snippets#Filename('$1_t', 'name')`};
 
 snippet enum
+	/*! \enum $1
+	 * 
+	 *  ${2:Detailed description} 	
+	 */
```

Comment:
As I mentioned above, I would propose to add d-prefixed versions with documentation section.